src/app/core/export/download-service/download.service.ts

```diff
@@ -90,17 +91,59 @@ export class DownloadService {
    * @returns string a valid CSV string of the input data
    */
   async createCsv(data: any[]): Promise<string> {
-    // Collect all properties because papa only uses the properties of the first object
+    let entityConstructor: any;
+
+    if (data.length > 0 && typeof data[0]?.getConstructor === 'function') {
+        entityConstructor = data[0].getConstructor();
+    }
     const keys = new Set<string>();
     data.forEach((row) => Object.keys(row).forEach((key) => keys.add(key)));
 
     data = data.map(transformToReadableFormat);
+   
+    if (!entityConstructor) {
+      return this.papa.unparse(data, {
+        quotes: true,
+        header: true,
+        newline: DownloadService.SEPARATOR_ROW,
+        columns: [...keys],
+      });
+    }
+
+    const result = this.exportFile(data, entityConstructor);
+    return result;
+  }
+
+  exportFile(data: any[], entityConstructor: { schema: any; }) {
+    const entitySchema = entityConstructor.schema;
+    const columnLabel = new Map<string, EntitySchemaField>();
+    
+    entitySchema.forEach((value: { label: string }, key: string) => {
+      if (value.label) columnLabel[key] = value.label;
+    });
+  
+    const exportEntities = data.map((item) => {
+      for (const key in item) {
+        if (!columnLabel.hasOwnProperty(key)) delete item[key];
```

Comment:
I think the `delete item[key]` creates the side effect of changing the original data array elements. It's probably better to work on a copy of the item instead.
this could otherwise affect the data that we work with within the app's list as well.

Reply:
I have updated it.

Reply:
thanks :slightly_smiling_face: